### ISSUE TITLE
Edge cases when Annotated is wrapping special type forms

### ISSUE THREAD

**Issue by GBeauregard** [microsoft/pyright#2954]
Note: if you are reporting a wrong signature of a function or a class in the standard library, then the typeshed tracker is better suited for this report: https://github.com/python/typeshed/issues.

**Describe the bug**
While working on the test suite for [bpo-46511](https://bugs.python.org/issue46511), I noticed some odd false positives on the lines with `Annotated` annotations.

**To Reproduce**
```python
from dataclasses import dataclass, fields
from typing import Annotated, ClassVar

@dataclass
class C:
    x: int
    y: int = 10
    z: ClassVar[int]                                    = 1000
    w: ClassVar[int]                                    = 2000
    t: ClassVar[int]                                    = 3000
    s: ClassVar                                         = 4000
    a: Annotated[ClassVar, 'meta']                      = 5000
    b: Annotated[ClassVar[int], 'meta']                 = 6000
    c: Annotated['ClassVar', 'meta']                    = 7000
    d: Annotated[Annotated[ClassVar, 'meta'], 'meta']   = 8000
    e: Annotated['Annotated[ClassVar, "meta"]', 'meta'] = 9000

c = C(5)
assert len(fields(C)) == 2
assert len(C.__annotations__) == 11
```

```console
/home/gregory/Downloads/testcase/pytest/antest/ntest.py
  /home/gregory/Downloads/testcase/pytest/antest/ntest.py:12:18 - error: "Protocol" cannot be used as a type argument
  /home/gregory/Downloads/testcase/pytest/antest/ntest.py:14:18 - error: "Protocol" cannot be used as a type argument
  /home/gregory/Downloads/testcase/pytest/antest/ntest.py:15:28 - error: "Protocol" cannot be used as a type argument
  /home/gregory/Downloads/testcase/pytest/antest/ntest.py:15:18 - error: "Protocol" cannot be used as a type argument
  /home/gregory/Downloads/testcase/pytest/antest/ntest.py:16:29 - error: "Protocol" cannot be used as a type argument
  /home/gregory/Downloads/testcase/pytest/antest/ntest.py:16:18 - error: "Protocol" cannot be used as a type argument
  /home/gregory/Downloads/testcase/pytest/antest/ntest.py:12:59 - error: Expression of type "Literal[5000]" cannot be assigned to declared type "ClassVar"
    "Literal[5000]" is incompatible with "ClassVar" (reportGeneralTypeIssues)
  /home/gregory/Downloads/testcase/pytest/antest/ntest.py:14:59 - error: Expression of type "Literal[7000]" cannot be assigned to declared type "ClassVar"
    "Literal[7000]" is incompatible with "ClassVar" (reportGeneralTypeIssues)
  /home/gregory/Downloads/testcase/pytest/antest/ntest.py:15:59 - error: Expression of type "Literal[8000]" cannot be assigned to declared type "ClassVar"
    "Literal[8000]" is incompatible with "ClassVar" (reportGeneralTypeIssues)
  /home/gregory/Downloads/testcase/pytest/antest/ntest.py:16:59 - error: Expression of type "Literal[9000]" cannot be assigned to declared type "ClassVar"
    "Literal[9000]" is incompatible with "ClassVar" (reportGeneralTypeIssues)
10 errors, 0 warnings, 0 informations
Completed in 0.632sec
```

**Expected behavior**
The code type checks

**VS Code extension or command-line**
pyright 1.1.216 Linux commandline

**Comment by erictraut** [comment]
These non-type type annotations (including `ClassVar`, `Final`, `InitVar` and `Annotated`) are a major pain. I really with these would have never been added to the language. They introduce so many special cases.

I've fixed the problems with `ClassVar` within an `Annotated`.

I'm not going to support quoted `"ClassVar"` or `"Annotated"` type annotations. This is just way too much special casing, and there's no valid reason for ever forward-declaring that part of a type annotation. You can always quote an inner part of the type (a type argument to `ClassVar` or `Annotated`) if a forward declaration is required.

**Comment by GBeauregard** [comment]
I agree with you re: forward-declaring that part of the annotation so it's not a big deal to not support on the type checker side of things.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.217, which I just published. It will also be included in the next release of pylance.
